src/components/QueryEditor/useInitQuery.ts

```diff
@@ -0,0 +1,58 @@
+import { Group, XrayQuery, XrayQueryType } from '../../types';
+import { XrayDataSource } from '../../DataSource';
+import { useEffect } from 'react';
+import { dummyAllGroup } from './constants';
+
+/**
+ * Inits the query on mount or on datasource change.
+ */
+export function useInitQuery(
+  query: XrayQuery,
+  onChange: (value: XrayQuery) => void,
+  groups: Group[],
+  dataSource: XrayDataSource
+) {
+  useEffect(() => {
+    // We assume here the "Default" group is always there.
+    const defaultGroup = groups.find((g: Group) => g.GroupName === 'Default')!;
```

Comment:
This is just an idea, feel free to ignore.

from here on `defaultGroup` is of type `Group | undefined` (omitting the `!` at the end) even though we know the dafault group is always present.

what if instead we do:
```ts
function assert(condition: any, msg?: 'Assertion Failed'): asserts condition {
  if (!condition) {
    throw new Error(msg);
  }
}
``` 
and here:

```ts
const defaultGroup = groups.find((g) => g.GroupName === 'Default');
assert(!!defaultGroup);
```

it's basically (almost) the same thing but I feel it's clearer.

What do you think?

Reply:
Is the `asserts condition` in `on Failed'): asserts condition {` some typescript thing? Can you link to docs cause I missed this if it's something new.

So not sure what you mean by `from here on defaultGroup is of type Group | undefined (omitting the ! at the end) `, the type should actually be `Group` because of the `!`.

So is this just to have more explicit assert instead of the `!` operator?

Reply:
Gonna merge this as I based some features on this refactor, but we can still continue this.

Reply:
you can find docs here: https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#assertion-functions

> the type should actually be Group because of the `!`.

true, type-wise they are perfectly identical.
> So is this just to have more explicit assert instead of the `!` operator?

Yes. This is not really needed, but I think is a great advantage because of how explicit it is.

The point is that using assertion functions over the non-null operator is that you know immediately if it breaks, while the non-null only works at compile time. the former will throw a big error on the console, the latter may produce hard-to-debug runtime side-effects (ie. if by any chance we'll change the default groupName to 'Default Group' in the future).